ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/CredentialExpirationCard.tsx

```diff
@@ -0,0 +1,90 @@
+import { TableComposable, Tbody, Td, Tr } from '@patternfly/react-table';
+
+import { Cluster } from 'types/cluster.proto';
+
+import { getCertificateExpirationCounts } from './ClustersHealth.utils';
+import ClustersHealthCardHeader from './ClustersHealthCardHeader';
+import {
+    TdDegraded,
+    TdHealthy,
+    TdTotal,
+    TdUnavailable,
+    TdUninitialized,
+    TdUnhealthy,
+    TheadClustersHealth,
+} from './ClustersHealthTable';
+
+const dataLabelHealthy = '\u2265 30 days'; // greater than or equal to
+const dataLabelUnhealthy = '< 7 days';
+const dataLabelDegraded = '< 30 days';
+
+export type CredentialExpirationCardProps = {
+    clusters: Cluster[];
+    currentDatetime: Date | null;
+    isFetching0: boolean;
+    requestErrorMessage: string;
+};
+
+function CredentialExpirationCard({
+    clusters,
+    currentDatetime,
+    isFetching0,
+    requestErrorMessage,
+}: CredentialExpirationCardProps): ReactElement {
+    const counts = currentDatetime
+        ? getCertificateExpirationCounts(clusters, currentDatetime)
+        : null;
+
+    /*
+     * Render card header without body:
+     * with spinner, if fetching first request
+     * with check mark, if counts are healthy: HEALTHY !== 0 and UNHEALTHY === 0 && DEGRADED === 0
+     *
+     * Render card body:
+     * for request error
+     * for table of counts if not healthy: HEALTHY === 0 || UNHEALTHY !== 0 || DEGRADED !== 0
+     */
+
+    /* eslint-disable no-nested-ternary */
+    return (
+        <Card isCompact>
+            <ClustersHealthCardHeader
+                counts={counts}
+                isFetching0={isFetching0}
+                title="Credential expiration"
+            />
+            {requestErrorMessage ? (
+                <CardBody>
+                    <Alert isInline variant="warning" title={requestErrorMessage} />
+                </CardBody>
+            ) : counts !== null &&
+              (counts.HEALTHY === 0 || counts.UNHEALTHY !== 0 || counts.DEGRADED !== 0) ? (
+                <CardBody>
+                    <TableComposable variant="compact">
+                        <TheadClustersHealth
+                            dataLabelHealthy={dataLabelHealthy}
+                            dataLabelUnhealthy={dataLabelUnhealthy}
+                            dataLabelDegraded={dataLabelDegraded}
+                        />
+                        <Tbody>
+                            <Tr>
+                                <Td>Clusters</Td>
+                                <TdHealthy counts={counts} dataLabel={dataLabelHealthy} />
+                                <TdUnhealthy counts={counts} dataLabel={dataLabelUnhealthy} />
+                                <TdDegraded counts={counts} dataLabel={dataLabelDegraded} />
+                                <TdUnavailable counts={counts} />
+                                <TdUninitialized counts={counts} />
+                                <TdTotal clusters={clusters} />
```

Comment:
What are the considerations of passing all the counts to each different type of cell, instead of passing the count each cell needs?

Reply:
Aha, your question pointed me to obsolete code from first draft!

`TdHealthy` component still has `count !== 0 && counts.UNHEALTHY === 0 && counts.DEGRADED === 0` to render green `CheckCircleIcon` but now the table never appears if healthy.

Therefore, I will change from `counts` to `count` prop.